### PR TITLE
Visitor can register an account

### PR DESCRIPTION
**Pull request by johnnysee**
PT: https://www.pivotaltracker.com/story/show/179716951

## User Story

```
As a user,
In order to be able to add product to an order
I would like to be able to register an account
```

## Changes Proposed

- Added a route to Signup.jsx
- Create a authentication function
- Create a register form
- Added email validation
- Semantic-UI styling to form and buttons
- Create and format fixtures for stubbing
- Email validation for registration form

## What we learned working on this feature
- Usefulness of React Semantic-UI Documentation
- Regex in React email validation
- Combined Vanilla JavaScript with Semantic-UI 

## Packages/Gems added
None.

## Screenshots/Videos

Screenshot 2021-10-02 at 15 30 11

Screenshot 2021-10-02 at 15 35 55

Screenshot 2021-10-02 at 15 36 04

### CODE REVIEW

File: cypress/integration/userCanCreateAnAccount.feature.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    cy.get("[data-cy=registration-message]").should(
+      "contain",
+      "Registration successful"
+    );
+  });
+
+  describe("Email validation", () => {
+    it("is expected not to display error with valid email address", () => {
+      cy.get("[data-cy=email-input]").type("user@email.com");
+      cy.get("#form-input-control-error-email-error-message").should(
```
Comment (tochman) [inline cypress/integration/userCanCreateAnAccount.feature.js]: Please stick to the `data-cy` way of finding elements

File: cypress/fixtures/userRegistrationResponse.json [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{
+  "id": 78,
+  "provider": "email",
+  "uid": "user@email.com",
+  "allow_password_change": false,
+  "email": "user@email.com",
+  "created_at": "2021-10-01T13:57:29.960Z",
+  "updated_at": "2021-10-01T13:57:29.968Z"
+}
```
Comment (tochman) [inline cypress/fixtures/userRegistrationResponse.json:12]: This fixture is still not what the server will respond with

File: src/components/Signup.jsx [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+  const handleSubmit = (evt) => {
+    evt.preventDefault();
+    axios({
+      method: "post",
+      url: "https://slowfood.heroku.com/api/auth/",
+      params: {
+        email: userEmail,
+        password: userPassword,
+        password_confirmation: userConfirmPassword,
+        confirm_success_url: "placeholder",
```
Comment (tochman) [inline src/components/Signup.jsx]: Not sure you need this param.

File: src/components/Signup.jsx [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+      },
+    }).then(setOpen(true));
+  };
+
+  const validateEmail = (email) => {
+    const re =
+      /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+    setValidEmail(re.test(String(email).toLowerCase()));
+  };
```
Comment (tochman) [inline src/components/Signup.jsx:28]: I would extract this to a module

Comment (maxarvid) [reply src/components/Signup.jsx:28]: We have extracted the function to a module.

File: cypress/integration/userCanCreateAnAccount.feature.js (hunk 1 above)
Comment (johnnysee) [reply cypress/integration/userCanCreateAnAccount.feature.js]: Noticed and done!

File: cypress/fixtures/userRegistrationResponse.json (hunk 2 above)
Comment (johnnysee) [reply cypress/fixtures/userRegistrationResponse.json:12]: Noticed and done!

File: src/components/Signup.jsx (hunk 3 above)
Comment (maxarvid) [reply src/components/Signup.jsx]: We have removed the `confirm_success_url` param.

**Pull request merged**